src/txmempool.h

```diff
@@ -127,6 +149,11 @@ class CTxMemPoolEntry
     CAmount GetModFeesWithAncestors() const { return nModFeesWithAncestors; }
     int64_t GetSigOpCostWithAncestors() const { return nSigOpCostWithAncestors; }
 
+    const Parents& GetMemPoolParentsConst() const { return m_parents; }
+    const Children& GetMemPoolChildrenConst() const { return m_children; }
+    Parents& GetMemPoolParents() const { return m_parents; }
+    Children& GetMemPoolChildren() const { return m_children; }
```

Comment:
nit: It seems redundant to offer two definitions for these getters - constness can be enforced by the target variable, no?

Reply:
nope it can't really -- e.g., if I change the type of the local binding it will automatically pick the correct const/non const version, but I want to call enforcing that it can't be used non-const.

see begin/cbegin https://en.cppreference.com/w/cpp/container/vector/begin which exists for the same purpose.

Reply:
Ok I took a crack at making an example of this but found it conflicted with the `indexed_transaction_set` constraints, which force const on the stored value. So this is fine by me.